app-frontend/src/app/pages/library/projects/detail/projectScenes/projectScenes.controller.js

```diff
@@ -186,4 +189,37 @@ export default class ProjectScenesController {
             }
         );
     }
-}
+
+    toggleProjectNameEdit() {
+        if (this.isEditingProjectName) {
+            this.cancelProjectNameEdit();
+        } else {
+            this.startProjectNameEdit();
+        }
+    }
+
+    startProjectNameEdit() {
+        this.editedProjectName = this.project.name;
+        this.isEditingProjectName = true;
+    }
+
+    saveProjectNameEdit() {
+        this.isSavingProjectNameEdit = true;
+        this.isEditingProjectName = false;
+        this.project.name = this.editedProjectName;
+        this.projectService.updateProject(this.project).then(
+            () => {
+                this.isSavingProjectNameEdit = false;
+            },
+            (err) => {
+                this.$log.error('Error renaming project:', err);
```

Comment:
Great catch on the project name. Changes now show a spinner (in place of the edit button) while request is ongoing. The project name is cached and reverted in the case of an error.